### PR TITLE
Add package_name field to pip_install

### PR DESCRIPTION
**Pull request by eivindjahren**
Implement the repository field "pypi_package_name" to specify name of a package in pypi when installing via `pip_install`.

### CODE REVIEW

File: komodo/build.py [hunk 1]
```diff
@@ -115,17 +115,31 @@ def rsync(pkg, ver, pkgpath, data, prefix, *args, **kwargs):
...
+        "--prefix {}".format(prefix),
+        "--no-index",
+        "--no-deps",
+        "--ignore-installed",
+        "--cache-dir {}".format(dlprefix),
+        "--find-links {}".format(dlprefix),
+        kwargs.get("makeopts", ""),
     ]
 
     print('Installing {} ({}) from pip'.format(pkg, ver))
```
Comment (jondequinor) [inline komodo/build.py:145]:
```suggestion
    print('Installing {} ({}) from pip'.format(package_name, ver))
```

File: komodo/build.py [hunk 2]
```diff
@@ -115,17 +115,31 @@ def rsync(pkg, ver, pkgpath, data, prefix, *args, **kwargs):
...
-           kwargs.get('makeopts', ''),
+def pip_install(
+    pkg,
+    ver,
+    pkgpath,
+    data,
+    prefix,
+    dlprefix,
+    pip="pip",
+    package_name=None,
```
Comment (jondequinor) [inline komodo/build.py:126]: For `pip`, `pkg == package_name`, so drop this? Use `pkg` from this point on.

File: komodo/build.py [hunk 3]
```diff
@@ -162,58 +176,79 @@ def make(pkgfile, repofile, data,
...
+        "noop": noop,
+    }
 
     for pkg, path in zip(pkgorder, pkgpaths):
         ver = pkgs[pkg]
         current = repo[pkg][ver]
-        make = current['make']
+        make = current["make"]
         pkgpath = os.path.abspath(path)
+        package_name = pkgs.get("package_name", None)
```
Comment (jondequinor) [inline komodo/build.py:228]: I think we should only support `package_name` for `source: pypi`, so do an assertion here.

Basically, if `package_name`, the `source` must be `pypi`.

**Review by jondequinor — COMMENTED** [review]
There's a second install of all pip packages at the bottom of `cli.py`. You need to fix this there as well.

File: komodo/build.py (hunk 2 above)
Comment (eivindjahren) [reply komodo/build.py]: This has now been changed

File: komodo/build.py (hunk 3 above)
Comment (eivindjahren) [reply komodo/build.py]: Added

**Comment by jondequinor** [comment]
Could you add a unit test for this? We have some testing in place already, so should be easy to extend.

**Review by jondequinor — APPROVED** [review]
Good stuff, thank you!

**Review by jondequinor — APPROVED** [review]
🥳

**Pull request merged**